action-scripts/merge_pr_jsons.py

```diff
@@ -23,43 +22,96 @@
 }
 ''')
 
+CHAIN_IDS = {
+    1: "mainnet",
+    3: "ropsten",
+    42161: "arbitrum",
+    100: "gnosis",
+    137: "polygon",
+    10: "optimism",
+}
+
+# Place your BIPs json into this directory under BIPs/<TARGET_DIR_WITH_BIPS>
+TARGET_DIR_WITH_BIPS = "00merging"
+
+
+def _parse_bip_json(file_path: str, chain: int) -> Optional[dict]:
+    """
+    In case file was created within given date bounds and for given chain -
+    parse it and return the data
+    """
+    # Check if the file is a json file
+    if not file_path.endswith(".json"):
+        return None
+    try:
+        with open(file_path, "r") as json_file:
+            data = json.load(json_file)
+            # Check if the file is a dictionary, not a list
+            if not isinstance(data, dict):
+                return None
+            # Check if chain id is the same as the one we are looking for
+            if int(data["chainId"]) == int(chain):
+                return data
+    except JSONDecodeError:
+        return None
+
+
 def main():
-    changed_files = sys.argv[1].split()
-    date = datetime.utcnow()
-    # Filter the list of added files for json files
-    json_files = [filename for filename in changed_files if filename.endswith(".json") and "BIPs/BIP" in filename]
-    txmap = {}
-    # Extract the relevant information from the JSON file
-    for json_file in json_files:
-        if debug:
-            print(f"Processing {json_file}")
-        # Get the JSON file from the repository
-        with open(json_file, "r") as json_data:
-            data = json.load(json_data)
-        # Extract the relevant information from the JSON file
-        chain_id = data["chainId"]
-        if chain_id not in txmap.keys():
-            txmap[chain_id]={}
-        safe = data["meta"]["createdFromSafeAddress"]
-        print(safe)
-        if safe not in txmap[chain_id].keys():
-            txmap[chain_id][safe] = []
-        txmap[chain_id][safe] = [*txmap[chain_id][safe], *data["transactions"]]
-        print(txmap)
+    current_week = datetime.utcnow().strftime("%U")
+    current_year = datetime.utcnow().year
+    # get root directory of the project:
+    root_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```

Comment:
This is hard for me/anyone who isn't a real pythonista to understand.  Can you provide a comment?

Reply:
is this just pwd?  Why so many nested calls?

Reply:
Sure. When you call `os.path.abspath(__file__)` that would be full absolute path to the executed python file. Like:
```
os.path.abspath(__file__) == ''opt/multisig-ops/action-scripts/merge_pr_jsons.py''
```

Then you just want to move up 2 times, for that you are using `os.path.dirname`, and you end up in `opt/multisig-ops/`, which is a root directory of your project

Reply:
Comment added